src/tests/build.proj

```diff
@@ -483,6 +483,10 @@
       <GroupBuildCmd Condition="'$(TestBuildMode)' == 'nativeaot'">$(GroupBuildCmd) "/p:DefaultBuildAllTarget=BuildNativeAot"</GroupBuildCmd>
       <GroupBuildCmd Condition="'$(IlcMultiModule)' == 'true'">$(GroupBuildCmd) "/p:IlcMultiModule=true"</GroupBuildCmd>
       <GroupBuildCmd Condition="'$(BuildNativeAotFrameworkObjects)' == 'true'">$(GroupBuildCmd) "/p:BuildNativeAotFrameworkObjects=true"</GroupBuildCmd>
+
+      <!-- Set Ilc and the SDK patch for cross compilation -->
```

Comment:
Am I right in understanding this is added here because `src/tests/Directory.Build.targets` doesn't set the properties for WASM?

Reply:
I think you can remove the conditions from `src/tests/Directory.Build.targets` and get past the missing `IlcSdkPath`.

You then hit 
```
  E:\GitHub\runtimelab2\.dotnet\sdk\6.0.100\MSBuild.dll -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,E:\GitHub\runtimelab2\.dotnet\sdk\6.0.100\dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,E:\GitHub\runtimelab2\.dotnet\sdk\6.0.100\dotnet.dll -maxcpucount /maxcpucount -verbosity:m /bl:E:\GitHub\runtimelab2\artifacts\/log/Debug/InnerManagedTestBuild.3.binlog /nodeReuse:false /p:TargetArchitecture=wasm /p:Configuration=Debug /p:LibrariesConfiguration=debug /p:TargetOS=Browser /p:RuntimeOS= /p:RuntimeFlavor=coreclr /p:RuntimeVariant= /p:CLRTestBuildAllTargets= /p:__TestGroupToBuild=3 /p:__SkipRestorePackages=1 /p:DefaultBuildAllTarget=BuildNativeAot /t:Build E:\GitHub\runtimelab2\src\tests\build.proj
    XUnitWrapperGenerator -> E:\GitHub\runtimelab2\artifacts\tests\coreclr\Browser.wasm.Debug\Common\XUnitWrapperGenerator\XUnitWrapperGenerator\XUnitWrapperGenerator.dll
    Dataflow -> E:\GitHub\runtimelab2\artifacts\tests\coreclr\Browser.wasm.Debug\nativeaot\SmokeTests\Dataflow\Dataflow\Dataflow.dll
    Generating native code
    '"E:\GitHub\runtimelab2\artifacts\bin\coreclr\Browser.wasm.Debug\ilc\\ilc"' is not recognized as an internal or external command,
    operable program or batch file.
E:\GitHub\runtimelab2\artifacts\bin\coreclr\Browser.wasm.Debug\build\Microsoft.NETCore.Native.targets(292,5): error MSB3073: The command ""E:\GitHub\runtimelab2\artifacts\bin\coreclr\Browser.wasm.Debug\ilc\\ilc" @"E:\GitHub\runtimelab2\artifacts\tests\coreclr/obj/Browser.wasm.Debug/Managed/nativeaot\SmokeTests\Dataflow\Dataflow\native\Dataflow.ilc.rsp"" exited with code 9009. [E:\GitHub\runtimelab2\src\tests\nativeaot\SmokeTests\Dataflow\Dataflow.csproj] [E:\GitHub\runtimelab2\src\tests\build.proj]
```
Because:
![image](https://user-images.githubusercontent.com/2720041/210647550-7c73c2f6-9f59-4249-b58d-c23811830fa4.png)

Reply:
The way I imagine this should work is that when we are cross-compiling the tests (which should always be the case for WASM), the ILC path should be set to the appropriate "host" value.

We used to achieve this with the code in `src/tests/Common/Directory.Build.targets` that constructed the core root. I now see [#64738](https://github.com/dotnet/runtime/pull/64738) actually just deleted that code, so it doesn't work anymore.

One particular piece of experience that needs to work is compiling individual tests without going through `src/tests/build.proj`.

Reply:
I see we already have `CoreCLRCrossILCompilerDir` defined `liveBuilds.targets`, so it should be "straightforward" to adapt code here to use it I would hope.

Reply:
or let me look into that, thanks.

Reply:
Doesn't look like it gets set properly, will trace that
![image](https://user-images.githubusercontent.com/2720041/210652388-6ad5e4ca-2024-4624-aa7c-8dc47cca3b32.png)

Reply:
I think the value is right, just the location somewhat unusual. What I do not recall is part of which build is this cross-component, host or target.

Reply:
Hmm, so I looked again at this cross-arch stuff and it appeared simpler to me to keep doing what we've been doing with the core root construction code, i. e. pick the host compiler out of host's build.

The cross-arch setup looks inconvenient for our scenario where cross-compilation is the mainline workflow.

Reply:
I think `CORE_ROOT` is setup in run.py, is that what you had in mind?

Reply:
I basically meant that we should migrate what we did in `src/tests/Common/Directory.Build.targets` with `<!-- When cross-compiling tests (such as is the case for Wasm), we depends on the host's ILC being built. -->` to here (and delete from there as it is now evidently dead code).

Reply:
Thanks, I see better now.  `src/tests/Common/Directory.Build.targets` would be ok perhaps, but the build fails before it gets a chance to run.  The target we want from there is `CopyDependencyToCoreRoot` however that ultimately is `dependsOn` for `GenerateLayout`.  That is a `TestBuildSteps` for `TestBuild`, the entry target, but its after `ManagedBuild` which is the failing step.  I tried to set up another dependency to `CopyDependencyToCoreRoot`:
```
  <Target Name="ManagedBuild"
      DependsOnTargets="CopyDependencyToCoreRoot;BuildManagedTestGroups"
```
 but it runs into problems because the msbuild properties are not inherited to `src/tests/Common/Directory.Build.targets`, the `RepoRoot` is not set, and `<Import Project="$(RepoRoot)eng/liveBuilds.targets" />` fails.  It might be possible to make this approach work, but I left it there.

Have moved the copy into a new `Target` in `src\tests\Directory.Build.targets`

Reply:
I don't think there is a need to move the core root construction code here, just the path setting.

I. e. before what happened was:
```
// 1) Core root is constructed with the paths / copy target
// 2) Compiler from core root is used when running the tests
```
Now what should happen is:
```
// 1) Compiler from the path we provide is used during the build
```
I. e. there is no more copying of ILC to the core root directory.

Reply:
```
  <Target Name="NativeAotWasmCopyDependencyToCoreRoot" BeforeTargets="IlcCompile" Condition="'$(TestBuildMode)' == 'nativeaot' and '$(TargetArchitecture)' == 'wasm'">

    <!-- Set IlcToolsPath for cross compilation -->
    <PropertyGroup>
      <IlcToolsPath>$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'artifacts', 'bin', 'coreclr', '$(HostOS).$(BuildArchitecture).$(CoreCLRConfiguration)\ilc'))</IlcToolsPath>
    </PropertyGroup>
    <Error Condition="!Exists('$(IlcToolsPath)')"
           Text="'$(IlcToolsPath)' does not exist. Subsets clr.jit, clr.wasmjit and clr.tools for $(HostOS) $(BuildArchitecture) in $(CoreCLRConfiguration) configuration must be built before building this project." />
  </Target>
```
Added this target which is passing in CI now.

Reply:
hold on, let me rename it

Reply:
The name's not quite right but we can fix it separately.

Reply:
ok, async human communication, beep.  I changed it.  @jkotas I guess if it it passes this time, then it can go, thanks

Reply:
Looks like a need to also fix a conflict